src/store/modules/segmentationModule.js

```diff
@@ -0,0 +1,1411 @@
+ * @param  {number} [segmentIndex]     The segment index.
+ * @returns {Object|Object[]}          A metadata object or an array of
+ *                                     metadata objects.
+ */
+function getMetadata(elementOrEnabledElementUID, labelmapIndex, segmentIndex) {
+  const element = _getEnabledElement(elementOrEnabledElementUID);
+
+  if (!element) {
+    return;
+  }
+
+  const stackState = getToolState(element, 'stack');
+  const stackData = stackState.data[0];
+  const firstImageId = stackData.imageIds[0];
+
+  const brushStackState = state.series[firstImageId];
+
+  if (!brushStackState) {
+    logger.warn(`brushStackState is undefined`);
+
+    return;
+  }
+
+  if (labelmapIndex === undefined) {
+    labelmapIndex = brushStackState.activeLabelmapIndex;
+  }
+
+  if (!brushStackState.labelmaps3D[labelmapIndex]) {
+    logger.warn(`No labelmap3D of labelmap index ${labelmapIndex} on stack.`);
+
+    return;
+  }
+
+  const labelmap3D = brushStackState.labelmaps3D[labelmapIndex];
+
+  if (segmentIndex === undefined) {
+    return labelmap3D.metadata;
+  }
+
+  return labelmap3D.metadata[segmentIndex];
+}
+
+/**
+ * GetLabelmaps3D - Returns the `Labelmap3D` objects associated with the series displayed
+ * in the element, the `activeLabelmapIndex` and the `currentImageIdIndex`.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
+ * @returns {Object}              An object containing `Labelmap3D` objects,
+ *                                the `activeLabelmapIndex` amd the `currentImageIdIndex`.
+ */
+function getLabelmaps3D(elementOrEnabledElementUID) {
+  const element = _getEnabledElement(elementOrEnabledElementUID);
+
+  if (!element) {
+    return;
+  }
+
+  const stackState = getToolState(element, 'stack');
+  const stackData = stackState.data[0];
+
+  const firstImageId = stackData.imageIds[0];
+  const brushStackState = state.series[firstImageId];
+
+  let labelmaps3D;
+  let activeLabelmapIndex;
+
+  if (brushStackState) {
+    labelmaps3D = brushStackState.labelmaps3D;
+    activeLabelmapIndex = brushStackState.activeLabelmapIndex;
+  }
+
+  return {
+    labelmaps3D,
+    activeLabelmapIndex,
+    currentImageIdIndex: stackData.currentImageIdIndex,
+  };
+}
+
+/**
+ * GetAndCacheLabelmap2D - Returns the active `labelmap3D` and the `currentImageIdIndex`.
+ *                         If a labelmap does not get exist, creates a new one.
+ *                         generates a `labelmap2D` for the `currentImageIndex` if it
+ *                         does not yet exist.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
+ * @returns {Object}              The `Labelmap3D` and the currentImageIdIndex.
+ */
+function getAndCacheLabelmap2D(elementOrEnabledElementUID) {
```

Comment:
Is this private/internal?

Reply:
When/why would you use this?

Reply:
Let me write the docs and we can discuss haha, this is public api that stops you from having to deal with everything else. If you want to draw to a labelmap from any tool/utility, you call this and you'll be given back a `Labelmap3D` to draw on, with an appropriate `Labelmap2D` view created for the current image. Looking at in reflection I should return the `Labelmap2D` itself at surface level also. I'll get back to you.